src/test/test_list.c

```diff
@@ -0,0 +1,276 @@
+#include <setjmp.h>
```

Comment:
Need to add the following tests:

- [ ] List insert empty
- [ ] List insert with data
- [ ] List insert end empty
- [ ] List insert end with data